### PR TITLE
Split and rename src/boot

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -2344,8 +2344,9 @@ subdir('src/ask-password')
 subdir('src/backlight')
 subdir('src/battery-check')
 subdir('src/binfmt')
+subdir('src/bless')
```
Comment (YHNdnzj) [inline meson.build]: I'd still call this bless-boot, otherwise very cryptic

File: man/systemd-boot.xml [hunk 2]
```diff
@@ -651,7 +651,7 @@
 
     <para>
       <command>qemu-system-x86_64 <replaceable>[ ... ]</replaceable>
-      -kernel /usr/lib/systemd/boot/efi/systemd-bootx64.efi</command>
+      -kernel /usr/lib/systemd/boot/systemd-bootx64.efi</command>
```
Comment (YHNdnzj) [inline man/systemd-boot.xml]: I doubt we can do this. It's become part of well-known paths and might be hardcoded in various other projects.

Comment (daandemeyer) [reply man/systemd-boot.xml]: We definitely can't do this, mkosi relies on the path being /usr/lib/systemd/boot/efi

Comment (bluca) [reply man/systemd-boot.xml]: yeah we have scripts relying on this too

Comment (keszybz) [reply man/systemd-boot.xml]: Well, first of all, a compat symlink is provided, so the old path will still work. And yes, we can rename things with enough announcement and forethought. We shouldn't keep every historical naming decision.

**Comment by keszybz** [comment]
> Let's not merge this until #35057 is in please

This PR is cleanly renaming source files. So git should actually apply the two pull requests without conflict.

File: man/systemd-boot.xml (hunk 2 above)
Comment (YHNdnzj) [reply man/systemd-boot.xml]: But what benefit does such rename provide? Rearranging our internal subdir structure is nice, but this change creates unnecessary churn?

Comment (keszybz) [reply man/systemd-boot.xml]: The old directory makes little sense. There's an additional subdir inserted that is not used for anything.

Comment (keszybz) [reply man/systemd-boot.xml]: I see that the opensuse spec file hardcodes both directories and the build fails.

But OK, let's maybe do the internal rename first. The install location will require some coordination with distros, so it's probably better to do it separately.

File: meson.build (hunk 1 above)
Comment (keszybz) [reply meson.build]: Done.

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
is this intended for v257 or v258?

(i'd be fine with the former, as this doesn't touch code, just build system and file arrangement, and i think if the CI passes this should give us enough assurance this doesn't break things too badly.)

**Review by bluca — APPROVED** [review]
Moving things in the git repo is fine I think

**Review by yuwata — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
> Let's not merge this until #35057 is in please

@yuwata I guessed you missed this :(

**Comment by yuwata** [comment]
> > Let's not merge this until #35057 is in please
> 
> @yuwata I guessed you missed this :(

Ah, uh, sorry.

**Pull request merged**